GPU/Software/Rasterizer.cpp

```diff
@@ -1418,6 +1418,10 @@ void DrawTriangleSlice(
 					if (gstate.isModeThrough()) {
 						s = Interpolate(v0.texturecoords.s(), v1.texturecoords.s(), v2.texturecoords.s(), w0, w1, w2, wsum_recip);
 						t = Interpolate(v0.texturecoords.t(), v1.texturecoords.t(), v2.texturecoords.t(), w0, w1, w2, wsum_recip);
+
+						// For levels > 0, mipmapping is always based on level 0.  Simpler to scale first.
+						s *= 1.0f / (float)gstate.getTextureWidth(0);
```

Comment:
Yeah - I think we need a "prim state" for the softgpu with some cached values - such as this.

-[Unknown]